ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java

```diff
@@ -44,7 +44,7 @@ public class TransactionDecoder {
 
   @FunctionalInterface
   interface Decoder {
-    Transaction decode(RLPInput input);
+    Transaction decode(RLPInput input, final DecodeType decodeType);
```

Comment:
Can we rename DecodeType to DecodeContext? It is actually not describing the type being decoded, or the type being produced. It is rather used to indicate which decoding context we are in, and what rules to use. 

We've now got 2 dimensions to consider during transaction codec operations, might warrant a refactor soon.